webrender/src/renderer.rs

```diff
@@ -1908,6 +1908,16 @@ impl Renderer {
                                     ExternalImageSource::RawData(data) => {
                                         self.device.update_pbo_data(&data[offset as usize..]);
                                     }
+                                    ExternalImageSource::Invalid => {
+                                        // Create a local buffer to fill the pbo.
+                                        let bpp = texture.get_bpp();
+                                        let width = stride.unwrap_or(rect.size.width);
+                                        let total_size = width * rect.size.width * bpp;
+                                        // WR haven't support RGBAF32 format in texture_cache, so
+                                        // we use u8 type here.
+                                        let dummy_data: Vec<u8> = vec![255; total_size as usize];
+                                        self.device.update_pbo_data(&dummy_data);
```

Comment:
I'm not sure whether we have alignment problem for the glBufferData() call here.

Reply:
> Clients must align data elements consistently with the requirements of the client platform, with an additional base-level requirement that an offset within a buffer to a datum comprising N bytes be a multiple of N.

Reply:
@kvark 
I have read these from the document, but I'm not sure I understand the meaning here.
For modern，data alignment could be 1, 2, 4 and more. So, what's the "requirement of the client platform"? And could we set the alignment hint for vector's buffer allocation?

Reply:
I wasn't able to find a concrete answer to that question. It might be just bad wording on the specification side, as noted by [this SO answer](https://gamedev.stackexchange.com/a/54876/6426).
In rust, there is `repr(align = XXX)` attribute but it's experimental and doesn't work in stable channel...

My conclusion is that the alignment is only required for data that is going to be directly accessed by GPU and/or DMA operations, which your `glBufferData` one is *not*. This data gets copied by the driver in the appropriate place, so alignment requirements don't really apply to your data source.